mev-boost-rs/src/service.rs

```diff
@@ -48,13 +47,18 @@ fn parse_url(input: &str) -> Option<Url> {
 pub struct Service {
     host: Ipv4Addr,
     port: u16,
-    relays: Vec<Url>,
+    relays: Vec<RelayEndpoint>,
     network: Network,
 }
 
 impl Service {
     pub fn from(config: Config) -> Self {
-        let relays: Vec<Url> = config.relays.iter().filter_map(|s| parse_url(s)).collect();
+        let relays: Vec<RelayEndpoint> = config
+            .relays
+            .iter()
+            .filter_map(|s| parse_url(s))
+            .filter_map(|url| RelayEndpoint::try_from(url).ok())
```

Comment:
I would personally lean towards something like `Relay::try_from` for a `&str` impl and collapse the URL parsing logic here into the `RelayEndpoint::try_from` logic here but this is growing on me the more I sit with it

Reply:
I'm planning on keeping this as is if you're alright with that